ouroboros-network/src/Ouroboros/Network/Socket.hs

```diff
@@ -453,7 +465,11 @@ withServerNode tbl addr encodeData decodeData acceptVersion versions k =
           decodeData
           acceptVersion
           throwIO
-          (\_connAddr st -> pure $ AcceptConnection st versions)
+          (\connAddr st ->
+            pure $ AcceptConnection
+                    st
+                    (peeridFn (Socket.addrAddress addr) connAddr)
```

Comment:
`Socket.addrAddress addr` is incorrect, you should use `addr'` here since `addr` may not be fully specified.

Reply:
Right, good catch :)

Reply:
Fixed.